### PR TITLE
Add new validation mode: xerces

### PR DESCRIPTION
**Pull request by olabusayoT**
- update tdml schema
- add new tests using none attribute to ensure no limited validation happens with xerces validation

DAFFODIL-2896

### CODE REVIEW

File: daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/parsers/ElementCombinator1.scala [hunk 1]
```diff
@@ -199,7 +199,8 @@ abstract class ElementParserBase(
       if (pstate.processorStatus ne Success) return
 
       val shouldValidate = erd.isSimpleType &&
-        pstate.dataProc.isDefined && pstate.dataProc.value.validationMode != ValidationMode.Off
+        pstate.dataProc.isDefined && (pstate.dataProc.value.validationMode != ValidationMode.Off
+          && pstate.dataProc.value.validationMode != ValidationMode.Xerces)
```
Comment (jadams-tresys) [inline daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/parsers/ElementCombinator1.scala:203]: Is this correct here as it seems that shouldValidate will always be false if ValidationMode.Xerces.  Is there a reason we need it to be false when validating with Xerces here?

Also this might be another case where a match statement makes things more clear.

File: daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/parsers/SequenceChildBases.scala [hunk 2]
```diff
@@ -530,7 +530,9 @@ trait EndArrayChecksMixin {
     if (state.processorStatus eq Success) {
 
       val shouldValidate =
-        state.dataProc.isDefined && state.dataProc.value.validationMode != ValidationMode.Off
+        state.dataProc.isDefined &&
+          (state.dataProc.value.validationMode != ValidationMode.Off
+            && state.dataProc.value.validationMode != ValidationMode.Xerces)
```
Comment (jadams-tresys) [inline daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/parsers/SequenceChildBases.scala:535]: Same question here

File: daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala [hunk 3]
```diff
@@ -928,8 +928,14 @@ abstract class TestCase(testCaseXML: NodeSeq, val parent: DFDLTestSuite) {
...
-        if (validationMode == ValidationMode.Full) false
-        else if (defaultValidationMode == ValidationMode.Full) false
+        if (
+          validationMode == ValidationMode.Full
+          || validationMode == ValidationMode.Xerces
+        ) false
+        else if (
+          defaultValidationMode == ValidationMode.Full
+          || defaultValidationMode == ValidationMode.Xerces
+        ) false
```
Comment (jadams-tresys) [inline daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala:938]: I think might be clearer in a match case:
```
(validationMode, defaultValidationMode) match {
    case (ValidationMode.Full | ValidationMode.Xerces, _) => false
    case (_, ValidationMode.Full | ValidationMode.Xerces) => false
    case (_, _) if (optExpectedWarnings.isDefined) => false
    case (_, _) => true
}
```

**Review by jadams-tresys — COMMENTED** [review]
I may not be familiar with how the validation is handled, but I'm not seeing any code that is actual running the Xerces validation.  Is that going to be in a separate pull request or is that a completely external process?

**Comment by stevedlawrence** [comment]
There are a number of comments in DAFFODIL-2896, we may want to consider more substantial changes to how validation works.

File: daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/parsers/ElementCombinator1.scala (hunk 1 above)
Comment (olabusayoT) [reply daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/parsers/ElementCombinator1.scala:203]: Yes, this shouldValidate is what controls whether we do limited validation, so this condition is saying do limited validation when validation mode is limited and full!

**Comment by olabusayoT** [comment]
Closing this in favor of https://github.com/apache/daffodil/pull/1326
